app/store/sagas/clubs.js

```diff
@@ -1,18 +1,31 @@
 import { call, select, put, takeEvery } from "redux-saga/effects";
 
-import { SET_CURRENT_USER_ID } from "../constants/actionTypes";
+import {
+  SET_CURRENT_CLUB_ID,
+  SET_CURRENT_USER_ID
+} from "../constants/actionTypes";
 import { updateEntity, setEntity } from "../actions/entities";
 import { setCurrentClubID } from "../actions/data";
-import { token, getCurrentUserID } from "../state/appState/selectors";
+import {
+  token,
+  getCurrentUserID,
+  getCurrentClubID,
+  getCurrentClubMembers
+} from "../state/appState/selectors";
 import {
   listClubMembers,
   listClubActivities,
   filterClubAthletes
 } from "../services/clubs";
 import { getRankings } from "../services/activities";
 
-function* listActivities(clubID, membersIDs) {
+function* listActivities() {
+  const clubID = yield select(getCurrentClubID);
+  if (clubID === "loading") return;
```

Comment:
it s strange clubID === "loading"

Reply:
We did (you did) the same for all application loading state!